sycl/include/sycl/ext/oneapi/experimental/group_helpers_sorters.hpp

```diff
@@ -0,0 +1,87 @@
+//==------- group_helpers_sorters.hpp - SYCL sorters and group helpers -----==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/detail/group_sort_impl.hpp>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace ext {
+namespace oneapi {
+namespace experimental {
+
+// ---- group helpers
+template <typename Group, std::size_t Extent> class group_with_scratchpad {
+  Group g;
+  sycl::span<std::uint8_t, Extent> scratch;
+
+public:
+  group_with_scratchpad(Group g_, sycl::span<std::uint8_t, Extent> scratch_)
+      : g(g_), scratch(scratch_) {}
+  Group get_group() const { return g; }
+  sycl::span<std::uint8_t, Extent> get_memory() const { return scratch; }
+};
+
+// ---- sorters
+template <typename Compare = std::less<>> class default_sorter {
+  Compare comp;
+  std::uint8_t *scratch;
+  std::size_t scratch_size;
```

Comment:
This is very similar to my previous comment. A pair of a pointer to an array and its size is either range or span. Since we have span backported to SYCL 2020, can we use it here?

Reply:
We consider this option previously during the API discussion. The thing is that if we replaced this with sycl::span, we would need an additional template parameter for sorter (extent). However, extent is not relevant to the `default_sorter`, but to the parameter of its ctor.